planetiler-core/src/main/java/com/onthegomap/planetiler/FeatureMerge.java

```diff
@@ -180,28 +185,97 @@ public static List<VectorTile.Feature> mergeLineStrings(List<VectorTile.Feature>
           }
         }
         List<LineString> outputSegments = new ArrayList<>();
-        for (Object merged : merger.getMergedLineStrings()) {
-          if (merged instanceof LineString line && line.getLength() >= lengthLimit) {
-            // re-simplify since some endpoints of merged segments may be unnecessary
-            if (line.getNumPoints() > 2 && tolerance >= 0) {
-              Geometry simplified = DouglasPeuckerSimplifier.simplify(line, tolerance);
-              if (simplified instanceof LineString simpleLineString) {
-                line = simpleLineString;
-              } else {
-                LOGGER.warn("line string merge simplify emitted {}", simplified.getGeometryType());
-              }
-            }
-            if (buffer >= 0) {
-              removeDetailOutsideTile(line, buffer, outputSegments);
-            } else {
-              outputSegments.add(line);
-            }
+        var i = 0;
+        for (var line : merger.getMergedLineStrings()) {
+          // TODO remove debug features comment
+          // Map<String, Object> attrs = new HashMap<>();
+          // attrs.put("idx", i++);
+          // result.add(feature1.copyWithNewGeometry(line.getStartPoint()).copyWithExtraAttrs(attrs));
+          // result.add(feature1.copyWithNewGeometry(line).copyWithExtraAttrs(attrs));
+          // attrs.put("end", "yes");
+          // result.add(feature1.copyWithNewGeometry(line.getEndPoint()).copyWithExtraAttrs(attrs));
+
+          if (buffer >= 0) {
+            removeDetailOutsideTile(line, buffer, outputSegments);
+          } else {
+            outputSegments.add(line);
+          }
+        }
+
+        if (!outputSegments.isEmpty()) {
+          outputSegments = sortByHilbertIndex(outputSegments);
+          Geometry newGeometry = GeoUtils.combineLineStrings(outputSegments);
+          result.add(feature1.copyWithNewGeometry(newGeometry));
+          // i = 0;
+          // for (var outputSegment : outputSegments) {
+          //   Map<String, Object> attrs = new HashMap<>();
+          //   attrs.put("idx", ++i);
+          //   result.add(feature1.copyWithNewGeometry(outputSegment).copyWithExtraAttrs(attrs));
+          // }
+        }
+      }
+    }
+    return result;
+  }
+
+  public static List<VectorTile.Feature> mergeLineStringsAllParams(List<VectorTile.Feature> features,
```

Comment:
We probably don't need this extra method any more. FeatureMerge should eventually have a builder API like LoopLineMerge has, but until then should keep the main entry point opinionated about what defaults it uses.

Reply:
Done